modules/imap/functions.php

```diff
@@ -1144,3 +1144,22 @@ function get_personal_ns($imap) {
     );
 }}
 
+/**
+ * @subpackage imap/functions
+ */
+if (!hm_exists('parse_sieve_config_host')) {
+function parse_sieve_config_host($host) {
+    $url = parse_url($host);
+    if (! $url || ! isset($url['host']) || ! isset($url['scheme']) ) {
```

Comment:
If the prefix scheme is optional, just require host here and actual parsing here (i.e. not false).